### PR TITLE
Fixed CI/CD, improvements and small refactorings

### LINKED ISSUE #168: Rename BasePytorchClassifier to BasePyTorchClassifier (capital T)
Self-explanatory. We should enforce consistency with the other classes using PyTorch.

https://github.com/pralab/secml-torch/blob/d4bf906cbcc8684a574f9644eeef31ae1d316719/src/secmlt/models/pytorch/base_pytorch_nn.py#L10

### LINKED ISSUE #174: Add reduction='none' automatically to loss functions when passed from torch.nn
The current attacks support using loss functions directly passed from pytorch, however they have reduction='mean' by default. This does not allow trackers to work properly. The reduction attribute should be fixed internally (and possibly outputting a warning when this is changed internally).

### PR DESCRIPTION
**Pull request by maurapintor**
* Now the loss reduction is automatically fixed if the attack passes the loss type instead of the instantiated loss #174.
* Renamed BasePytorchClassifier to BasePyTorchClassifier #168.
* Upgraded all checkout and setup-python actions + fixed setuptools issue with visdom logger #182.
* Added `llm` extra in documentation requirements.
* Removed unnecessary tests from coverage.
* Added api for invert manipulation for returning delta in domain-constrained input space #181.

### CODE REVIEW

**Review by zangobot — APPROVED** [review]
LGTM
